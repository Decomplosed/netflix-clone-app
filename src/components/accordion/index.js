import React from 'react';

export default function Accordion({ children, ...restProps }) {
  return <Container></Container>;
}
