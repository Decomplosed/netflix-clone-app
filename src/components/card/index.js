import React from 'react';

export const FeatureContext = createContext();
