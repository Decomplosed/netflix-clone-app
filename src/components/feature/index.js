import React from 'react';
import { Container, Title, SubTitle } from './styles/feature';

export default function Feature({ children, ...restProps }) {
  return <Container></Container>
}
