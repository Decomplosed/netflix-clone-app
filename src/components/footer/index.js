import React from 'react';

export default function Footer({ children, ...restProps }) {
  return <Container {...restProps}></Container>;
}
