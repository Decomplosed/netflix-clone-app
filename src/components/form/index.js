import React from 'react';
import {
  Container,
  Base,
  Error,
  Title,
  Text,
  TextSmall,
  Link,
  Input,
  Submit,
} from './styles/form';

export default function Form({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Form.Base = function FormBase({ children, ...restProps }) {
  return <Base {...restProps}>{children}</Base>;
};

Form.Error = function FormError({ children, ...restProps }) {
  return <Error {...restProps}>{children}</Error>;
};

Form.Title = function FormTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};
