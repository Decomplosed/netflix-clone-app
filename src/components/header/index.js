import React from 'react';

export default Header({bg = true, children, ...restProps}) {}