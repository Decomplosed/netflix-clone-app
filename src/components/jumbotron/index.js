import React from 'react';

export default Jumbotron = () => {
  return <p>I am jumbotron</p>;
};
