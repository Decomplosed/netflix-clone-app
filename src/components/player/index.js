import React, { useState, useContext, createContext } from 'react';
import ReactDOM from 'react-dom';
