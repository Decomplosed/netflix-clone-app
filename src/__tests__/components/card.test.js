import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Card } from '../../components';

const category = 'series';
const slideRows = [
  {
    title: 'Documentaries',
    data: [
      {
        genre: 'documentaries',
        maturity: '18',
        slug: 'tiger-king',
        description: 'Tiger King description',
        id: '07adcf19-517f-41b4-8f8b-04fee694bce1',
        title: 'Tiger King',
        docId: 'IcRxJBanzE7nmIp2GMGb',
      },
    ],
  },
  {
    title: 'Feel Good',
    data: [
      {
        title: 'Juno',
        genre: 'feel-good',
        description: 'Juno description',
        maturity: '0',
        id: 'a938b5a6-fe25-4a06-8050-353bc7146ccd',
        slug: 'juno',
        docId: '4JDgdf5vE0K3YrW9ZnbP',
      },
    ],
  },
];

describe('<Card />', () => {
  it('Renders the <Card /> with populated data', () => {
    const { container, getByText } = render(
      <Card.Group>
        {slideRows.map((slideItem) => (
          <Card key={`${category}-${slideItem.title.toLowerCase()}`}>
            <Card.Title>{slideItem.title}</Card.Title>
            <Card.Entities>
              {slideItem.data.map((item) => (
                <Card.Item key={item.docId} item={item}>
                  <Card.Image
                    src={`/images/${category}/${item.genre}/${item.slug}/small.jpg`}
                  />
                </Card.Item>
              ))}
            </Card.Entities>
          </Card>
        ))}
      </Card.Group>,
    );
  });
});
