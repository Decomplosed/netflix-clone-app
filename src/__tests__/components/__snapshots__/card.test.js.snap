// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> Renders the <Card /> and clicks the card feature 1`] = `
<div
  class="card__Group-sc-17gu7hr-2 jDQCQI"
>
  <div
    class="card__Container-sc-17gu7hr-1 tFHzL"
  >
    <p
      class="card__Title-sc-17gu7hr-0 gOzYkP"
    >
      Documentaries
    </p>
    <div
      class="card__Entities-sc-17gu7hr-5 XmzIk"
    >
      <div
        class="card__Item-sc-17gu7hr-8 bdZSEj"
        data-testid="tiger-king-item-feature"
      >
        <img
          class="card__Image-sc-17gu7hr-7 gKfrHE"
          src="/images/series/documentaries/tiger-king/small.jpg"
        />
        <div
          class="card__Meta-sc-17gu7hr-6 hUPbVM"
        >
          <p
            class="card__SubTitle-sc-17gu7hr-3 dxnqzj"
          >
            Tiger King
          </p>
          <p
            class="card__Text-sc-17gu7hr-4 ddmioa"
          >
            Tiger King description
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="card__Container-sc-17gu7hr-1 tFHzL"
  >
    <p
      class="card__Title-sc-17gu7hr-0 gOzYkP"
    >
      Feel Good
    </p>
    <div
      class="card__Entities-sc-17gu7hr-5 XmzIk"
    >
      <div
        class="card__Item-sc-17gu7hr-8 bdZSEj"
        data-testid="juno-item-feature"
      >
        <img
          class="card__Image-sc-17gu7hr-7 gKfrHE"
          src="/images/series/feel-good/juno/small.jpg"
        />
        <div
          class="card__Meta-sc-17gu7hr-6 hUPbVM"
        >
          <p
            class="card__SubTitle-sc-17gu7hr-3 dxnqzj"
          >
            Juno
          </p>
          <p
            class="card__Text-sc-17gu7hr-4 ddmioa"
          >
            Juno description
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Card /> Renders the <Card /> with populated data 1`] = `
<div
  class="card__Group-sc-17gu7hr-2 jDQCQI"
>
  <div
    class="card__Container-sc-17gu7hr-1 tFHzL"
  >
    <p
      class="card__Title-sc-17gu7hr-0 gOzYkP"
    >
      Documentaries
    </p>
    <div
      class="card__Entities-sc-17gu7hr-5 XmzIk"
    >
      <div
        class="card__Item-sc-17gu7hr-8 bdZSEj"
      >
        <img
          class="card__Image-sc-17gu7hr-7 gKfrHE"
          src="/images/series/documentaries/tiger-king/small.jpg"
        />
        <div
          class="card__Meta-sc-17gu7hr-6 hUPbVM"
        >
          <p
            class="card__SubTitle-sc-17gu7hr-3 dxnqzj"
          >
            Tiger King
          </p>
          <p
            class="card__Text-sc-17gu7hr-4 ddmioa"
          >
            Tiger King description
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="card__Container-sc-17gu7hr-1 tFHzL"
  >
    <p
      class="card__Title-sc-17gu7hr-0 gOzYkP"
    >
      Feel Good
    </p>
    <div
      class="card__Entities-sc-17gu7hr-5 XmzIk"
    >
      <div
        class="card__Item-sc-17gu7hr-8 bdZSEj"
      >
        <img
          class="card__Image-sc-17gu7hr-7 gKfrHE"
          src="/images/series/feel-good/juno/small.jpg"
        />
        <div
          class="card__Meta-sc-17gu7hr-6 hUPbVM"
        >
          <p
            class="card__SubTitle-sc-17gu7hr-3 dxnqzj"
          >
            Juno
          </p>
          <p
            class="card__Text-sc-17gu7hr-4 ddmioa"
          >
            Juno description
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
